cf-upgrade/log.c

```diff
@@ -47,11 +48,14 @@ static char *prepare_message(char *format, va_list args)
     int message_size = 0;
     char timestamp[] = "YYYY/MM/DD HH:MM:SS";
     char buffer[MAX_LOG_ENTRY_SIZE];
-    time_t now = time(NULL);
-    struct tm *now_tm = gmtime(&now);
+    struct tm now;
+    {
+        time_t now_seconds = time(NULL);
+        gmtime_r(&now_seconds, &now);
+    }
```

Comment:
This nested block is really confusing here. It looks to me like if the `struct` was initialized using itself at the first glance. Please add a blank line between the declaration and the block or get rid of the nested block.